### ISSUE TITLE
Strings exercises should come earlier

### ISSUE THREAD

**Issue by mattrobineau** [rust-lang/rustlings#716]
In structs1, the user is told to create some structs which take strings (according to asserts). 

Being new to rust I tried:
```
struct ColorClassicStruct {
    name: String,
    hex: String,
}

// ..snip..

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn classic_c_structs() {
        // TODO: Instantiate a classic c struct!
        let green = ColorClassicStruct {
            name: "green",
            hex: "#00FF00",
       }

        assert_eq!(green.name, "green");
        assert_eq!(green.hex, "#00FF00");
    }
    // ..snip..
}
```

I checked the documentation for how to create/use strings to solve.

Could the strings sections be moved ahead of any exercises that use strings?

**Comment by Zerotask** [comment]
@mattrobineau 
Yes, I agree with you. For total beginners this is a gap. I guess the reason it comes later is that it's oriented at The Book (chapter 8) and Rust by example (chapter19).
Personally I'd teach the data types including variable declaration right at the beginning and since strings are something important, I'd also count them to data types.

**Comment by mattrobineau** [comment]
> @mattrobineau> Yes, I agree with you. For total beginners this is a gap. I guess the reason it comes later is that it's oriented at The Book (chapter 8) and Rust by example (chapter19).

I wasn't aware that this follows The Book, in that case, another option could be to use i32 in the structs examples instead of strings.

**Comment by chris-english** [comment]
Also completely new and reviewing up to Chapter 5, and given what's involved to pass agree with @mattrobineau for the purposes of introducing _struct_, i32 would be a better choice.

// structs1.rs
// Address all the TODOs to make the tests pass!

// I AM NOT DONE

struct ColorClassicStruct {
    name: String,
    hex: String,
}
//chapters 1-5 don't get us here: https://users.rust-lang.org/t/in-a-struct-why-must-i-use-str-and-a-lifetime/44482/8 as we're probably not appreciators of idiomatic Rust, yet.

impl ColorClassicStruct {
    fn new(name: &str, hex: &str) -> Self {
        Self {
            name: name.into(),
            hex: hex.into(),
        }
    }
}

struct ColorTupleStruct(String, String);

#[derive(Debug)]
struct UnitStruct;

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn classic_c_structs() {
        // TODO: Instantiate a classic c struct!
        let green = ColorClassicStruct::new("green", "#00FF00");
            //name: "green",
            //hex:  "#00FF00",
        

        assert_eq!(green.name, "green");
        assert_eq!(green.hex, "#00FF00");
    }

    #[test]
    fn tuple_structs() {
        // TODO: Instantiate a tuple struct!
         let green = ColorTupleStruct("green".to_string(), "#00FF00".to_string()); //nor to &.to_string methods, I think.

        assert_eq!(green.0, "green");
        assert_eq!(green.1, "#00FF00");
    }

    #[test]
    fn unit_structs() {
        // TODO: Instantiate a unit struct!
        let unit_struct = UnitStruct;
        let message = format!("{:?}s are fun!", unit_struct);

        assert_eq!(message, "UnitStructs are fun!");
    }
}

**Comment by diannasoreil** [comment]
Fixed in https://github.com/rust-lang/rustlings/commit/19bec503998c3c5618254015bada63b44a713f0e.
